src/addrman.cpp

```diff
@@ -810,6 +810,12 @@ std::vector<CAddress> AddrManImpl::GetAddr_(size_t max_addresses, size_t max_pct
     AssertLockHeld(cs);
 
     size_t nNodes = vRandom.size();
+    if (network) {
+        auto it = m_network_counts.find(*network);
+        if (it == m_network_counts.end()) return {};
+        const auto counts{it->second};
+        nNodes = counts.n_new + counts.n_tried;
```

Comment:
I think that this changes behavior in theory:
If both a network and `max_pct` are specified, it will now return `max_pct` of nodes from that network (before, `max_pct` would apply to all nodes).
In practice, that doesn't really matter because we specify either the one or the other in existing calls.

Reply:
Yes, I noticed it but ignored because in practice it won't happen. Perhaps worth adding an assertion?

Reply:
I think it is fine without an assert because nothing bad will happen if both are provided.